x-pack/plugin/inference/src/main/java/org/elasticsearch/xpack/inference/services/voyageai/rerank/VoyageAIRerankTaskSettings.java

```diff
@@ -0,0 +1,190 @@
+                ? requestTaskSettings.getReturnDocuments()
+                : originalSettings.getReturnDocuments(),
+            requestTaskSettings.getTruncation() != null ? requestTaskSettings.getTruncation() : originalSettings.getTruncation()
+
+        );
+    }
+
+    public static VoyageAIRerankTaskSettings of(Integer topKDocumentsOnly, Boolean returnDocuments, Boolean truncation) {
+        return new VoyageAIRerankTaskSettings(topKDocumentsOnly, returnDocuments, truncation);
+    }
+
+    private final Integer topKDocumentsOnly;
+    private final Boolean returnDocuments;
+    private final Boolean truncation;
+
+    public VoyageAIRerankTaskSettings(StreamInput in) throws IOException {
+        this(in.readOptionalInt(), in.readOptionalBoolean(), in.readOptionalBoolean());
+    }
+
+    public VoyageAIRerankTaskSettings(
+        @Nullable Integer topKDocumentsOnly,
+        @Nullable Boolean doReturnDocuments,
+        @Nullable Boolean truncation
+    ) {
+        this.topKDocumentsOnly = topKDocumentsOnly;
+        this.returnDocuments = doReturnDocuments;
+        this.truncation = truncation;
+    }
+
+    @Override
+    public boolean isEmpty() {
+        return topKDocumentsOnly == null && returnDocuments == null && truncation == null;
+    }
+
+    @Override
+    public XContentBuilder toXContent(XContentBuilder builder, Params params) throws IOException {
+        builder.startObject();
+        if (topKDocumentsOnly != null) {
+            builder.field(TOP_K_DOCS_ONLY, topKDocumentsOnly);
+        }
+        if (returnDocuments != null) {
+            builder.field(RETURN_DOCUMENTS, returnDocuments);
+        }
+        if (truncation != null) {
+            builder.field(TRUNCATION, truncation);
+        }
+        builder.endObject();
+        return builder;
+    }
+
+    @Override
+    public String getWriteableName() {
+        return NAME;
+    }
+
+    @Override
+    public TransportVersion getMinimalSupportedVersion() {
+        return TransportVersions.VOYAGE_AI_INTEGRATION_ADDED;
+    }
+
+    @Override
+    public void writeTo(StreamOutput out) throws IOException {
+        out.writeOptionalInt(topKDocumentsOnly);
+        out.writeOptionalBoolean(returnDocuments);
+        out.writeOptionalBoolean(truncation);
+    }
+
+    @Override
+    public boolean equals(Object o) {
+        if (this == o) return true;
+        if (o == null || getClass() != o.getClass()) return false;
+        VoyageAIRerankTaskSettings that = (VoyageAIRerankTaskSettings) o;
+        return Objects.equals(topKDocumentsOnly, that.topKDocumentsOnly)
+            && Objects.equals(returnDocuments, that.returnDocuments)
+            && Objects.equals(truncation, that.truncation);
+    }
+
+    @Override
+    public int hashCode() {
+        return Objects.hash(truncation, returnDocuments, topKDocumentsOnly);
+    }
+
+    public static String invalidInputTypeMessage(InputType inputType) {
```

Comment:
Is this used? I think input type is only relevant for the embedding task type. If it's not used, let's remove it.